Uno.Gallery/Uno.Gallery.UWP/Views/SamplePages/GamepadSamplePage.xaml

```diff
@@ -0,0 +1,188 @@
+	Gamepad.GamepadRemoved += GamepadsChanged;
+
+	_timer.Interval = TimeSpan.FromMilliseconds(100);
+	_timer.Tick += OnGamepadReadingUpdate;
+	_timer.Start();
+
+	UpdateGamepads();
+}
+
+public void StopObservingGamepadChanges()
+{
+	Gamepad.GamepadAdded -= GamepadsChanged;
+	Gamepad.GamepadRemoved -= GamepadsChanged;
+
+	_timer.Stop();
+
+	AvailableGamepads.Clear();
+}
+
+
+private void OnGamepadReadingUpdate(object sender, object e)
+{
+	UpdateGamepads();
+	foreach(var gamepad in AvailableGamepads)
+	{
+		gamepad.Update();
+		gamepad.Position = AvailableGamepads.IndexOf(gamepad)+1;
+	}
+}
+
+private void GamepadsChanged(object sender, Gamepad e)
+{
+	UpdateGamepads();
+}
+
+private async void UpdateGamepads()
+{
+	await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
+	{
+		var existingGamepads = new HashSet<Gamepad>(AvailableGamepads.Select(g => g.Gamepad));
+
+		var gamepadsToRemove = existingGamepads.Except(Gamepad.Gamepads);
+		var gamepadsToAdd = Gamepad.Gamepads.Except(existingGamepads);
+
+		foreach(var gamepad in gamepadsToRemove)
+		{
+			var vmToRemove = AvailableGamepads.FirstOrDefault(g => g.Gamepad == gamepad);
+			AvailableGamepads.Remove(vmToRemove);
+		}
+
+		foreach(var gamepad in gamepadsToAdd)
+		{
+			var vmToAdd = new GamepadViewModel(gamepad);
+			AvailableGamepads.Add(vmToAdd);
+		}
+	});
+}
+
+
+public class GamepadViewModel : ViewModelBase
+{
+	public GamepadViewModel(Gamepad gamepad)
+	{
+		Gamepad = gamepad;
+	}
+
+	public int Position { get => GetProperty<int>(); set => SetProperty(value); }
+
+	public Gamepad Gamepad { get; }
+
+	public void Update()
+	{
+		var reading = Gamepad.GetCurrentReading();
+
+		Buttons = reading.Buttons.ToString("g");
+
+		RightThumbstickX = reading.RightThumbstickX.ToString("0.00");
+		RightThumbstickY = reading.RightThumbstickY.ToString("0.00");
+		LeftThumbstickX = reading.LeftThumbstickX.ToString("0.00");
+		LeftThumbstickY = reading.LeftThumbstickY.ToString("0.00");
+
+		LeftTrigger = reading.LeftTrigger.ToString("0.00");
+		RightTrigger = reading.RightTrigger.ToString("0.00");
+	}
+
+	public string Buttons { get => GetProperty<string>(); private set => SetProperty(value); }
+
+	public string RightThumbstickX { get => GetProperty<string>(); private set => SetProperty(value); }
+
+	public string RightThumbstickY { get => GetProperty<string>(); private set => SetProperty(value); }
+
+	public string LeftThumbstickX { get => GetProperty<string>(); private set => SetProperty(value); }
+
+	public string LeftThumbstickY { get => GetProperty<string>(); private set => SetProperty(value); }
+
+	public string LeftTrigger { get => GetProperty<string>(); private set => SetProperty(value); }
+
+	public string RightTrigger { get => GetProperty<string>(); private set => SetProperty(value); }
+}
+-->
+
+								<StackPanel.DataContext>
+									<samples:GamepadSamplePageViewModel />
+								</StackPanel.DataContext>
```

Comment:
Just an FYI, the `SamplePageAttribute` in the code-behind has a `DataType` property that you can use to setup the `DataContext` for the sample template. It'll expose the VM as `Data` within the sample `DataTemplate`

Example here: https://github.com/unoplatform/Uno.Gallery/blob/master/Uno.Gallery/Uno.Gallery.UWP/Views/SamplePages/EmailManagerSamplePage.xaml.cs#L27

Not necessary to make those changes in this PR unless you want to change it, just pointing it out for further PRs that you might submit :)